File: TestABAsset/Assets/Scripts/MyLib/Debug/ShowFPS.js
#pragma strict

// Use this for initialization
private var rect : Rect;
private var fontSize : int = 20;
function Start () {
	var ratioH : float = (Camera.main.pixelWidth / 480.0);
	fontSize = fontSize * ratioH;
	rect.x = 0;
	rect.width = 100 * ratioH;
	rect.height = 50 * ratioH;
	rect.y = Camera.main.pixelHeight - fontSize * 1.5;
	//Application.targetFrameRate = 30;
}

// Update is called once per frame
function Update () {
	UpdateTick();
}

function OnGUI()
{
	DrawFps();
}

private function DrawFps()
{
	if (mLastFps > 50)
	{
		GUI.color = Color.green;
	}
	else if (mLastFps > 40)
	{
		GUI.color = Color.yellow;
	}
	else
	{
		GUI.color = Color.red;
	}
	GUI.Label(rect, "<size="+fontSize.ToString()+">fps: " + mLastFps + "</size>");

}

private var mFrameCount : long = 0;
private var mLastFrameTime : long = 0;
static var mLastFps : long= 0;

private function UpdateTick()
{
	if (true)
	{
		mFrameCount++;
		var nCurTime : long = TickToMilliSec(System.DateTime.Now.Ticks);
		if (mLastFrameTime == 0)
		{
			mLastFrameTime = TickToMilliSec(System.DateTime.Now.Ticks);
		}

		if ((nCurTime - mLastFrameTime) >= 1000)
		{
			var fps : long = (mFrameCount * 1.0f / ((nCurTime - mLastFrameTime) / 1000.0f));

			mLastFps = fps;

			mFrameCount = 0;

			mLastFrameTime = nCurTime;
		}
	}
}

static function TickToMilliSec(tick : long)
{
	return tick / (10 * 1000);
}

File: TestABAsset/Assets/Scripts/ShowFPS.cs
using UnityEngine;
using UnityEngine.UI;
public class ShowFPS : MonoBehaviour
{
    int m_FrameCount = 0;//帧数;
    double m_AddupFrameTime; //累计时间

    private float m_FPS = 0;

    public Text showtext;//显示fps的文本
                         // Update is called once per frame
    void Update()
    {
        m_FrameCount++;
        m_AddupFrameTime += Time.deltaTime;

        if (m_FrameCount >= 30)
        {
            m_FPS = (float)(1 / (m_AddupFrameTime / 30f));
            m_AddupFrameTime = 0;
            m_FrameCount -= 30;
        }
        ShowFpsByText();
    }

    //显示fps
    private void ShowFpsByText()
    {
        if (showtext != null)
        {
            showtext.text = string.Format("FPS: {0}", m_FPS + 5);//和unity报出来的差了5帧，不知原因
        }
    }

    // void OnGUI()
    // {
    //     GUI.Label(new Rect(Screen.width / 2, 0, 100, 100), "FPS: " + (m_FPS + 5)); //和unity报出来的差了5帧，不知原因
    // }

}
